pymeasure/display/widgets/table_widget.py

```diff
@@ -0,0 +1,544 @@
+        return cols
+
+    def _get_row_column_set(self, result, r1, c1, r2, c2):
+        top = self.translate_to_global(result, r1, c1)
+        bottom = self.translate_to_global(result, r2, c2)
+        top_bottoms = []
+        for i in range(c1, c2 + 1):
+            top = self.translate_to_global(result, r1, i)
+            bottom = self.translate_to_global(result, r2, i)
+            top_bottoms.append(top + bottom)
+
+        return top_bottoms
+
+    def translate_to_local(self, row, col):
+        """ Translate from full table coordinate to single result coordinates """
+        columns = 0
+        for index, result in enumerate(self.results_list):
+            if col < (columns + result.columns):
+                break
+            columns += result.columns
+        return result, row, col - columns
+
+    def translate_to_global(self, result, row, col):
+        """ Translate from single result coordinates to full table coordinates """
+        columns = 0
+        for res in self.results_list:
+            if res == result:
+                break
+            columns += result.columns
+        return row, col + columns
+
+    @property
+    def horizontal_header(self):
+        size = len(self.results_list)
+        if size:
+            v = list(self.results_list[0].data.columns)
+            return v * size
+        else:
+            return []
+
+    def horizontal_header_decoration(self, section):
+        result, _, _ = self.translate_to_local(0, section)
+        pixelmap = QtGui.QPixmap(6, 6)
+        pixelmap.fill(result.color)
+        return pixelmap
+
+
+class Table(QtWidgets.QTableView):
+    """ Table format view of :class:`Experiment<pymeasure.display.manager.Experiment>`
+    objects
+
+    """
+
+    supported_formats = {
+        'csv': "CSV file (*.csv)",
+        'excel': "Excel file (*.xlsx)",
+        'html': "HTML file (*.html *.htm)",
+        'json': "JSON file (*.json)",
+        'latex': "LaTeX file (*.tex)",
+        'markdown': "Markdown file (*.md)",
+        'xml': "XML file (*.xml)",
+    }
+
+    def __init__(self, refresh_time=0.2, check_status=True,
+                 force_reload=False, by_column=True, parent=None):
+        super().__init__(parent)
+        self.force_reload = force_reload
+        if by_column:
+            model = PandasModelByColumn()
+        else:
+            model = PandasModelByRow()
+
+        if SORTING_ENABLED:
+            proxyModel = QtCore.QSortFilterProxyModel(self)
+            proxyModel.setSourceModel(model)
+            model = proxyModel
+
+            model.setSortRole(SORT_ROLE)
+        self.setModel(model)
+        self.horizontalHeader().setStyleSheet("font: bold;")
+        self.setSortingEnabled(True)
+        self.sortByColumn(-1, QtCore.Qt.SortOrder.AscendingOrder)
+        self.horizontalHeader().setSectionsMovable(True)
+        self.horizontalHeader().setSectionResizeMode(QtWidgets.QHeaderView.ResizeToContents)
```

Comment:
Done in commit https://github.com/pymeasure/pymeasure/pull/771/commits/5d400e759c0be400a61724489b6c0b49c03644ae.
I suspect you are using PyQt6, because I was not having this problem. In any case, your fixes are relevant and correct. Thank you